Assorted bond/react fixes

jrgissing opened the pull request:
**Summary**

1) Add support for per-atom masses when creating new atoms
2) Remove unnecessary check for communication cutoff when initiator atoms are directly bonded
3) bug fix for when reaction site has angles, but post-reaction template has none (same for dihedrals, impropers)
4) another improved check for valid reaction templates

Thanks to Mu-Hung Chang, @mkanski, and Iakovos Delasoudas, Christophe Vaillant

**Related Issue(s)**

none

**Author(s)**

Jake Gissinger

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

<!--Please state whether any changes in the pull request will break backward compatibility for inputs, and - if yes - explain what has been changed and why-->

**Implementation Notes**

<!--Provide any relevant details about how the changes are implemented, how correctness was verified, how other features - if any - in LAMMPS are affected-->

**Post Submission Checklist**

<!--Please check the fields below as they are completed **after** the pull request has been submitted. Delete lines that don't apply-->

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

<!--Put any additional information here, attach relevant text or image files, and URLs to external sites (e.g. DOIs or webpages)-->

jrgissing committed:
inserting atoms: correct logic for per-atom mass

jrgissing committed:
better way to list rxn counts

jrgissing committed:
remove unnecessary restriction

do not check for comm cutoff when initiator atoms are directly bonded

stanmoore1 commented:
@jrgissing this has been idle for a while, any update on this PR? (Let us know when this is ready for review by removing the "draft" marking).

jrgissing commented:
@stanmoore1 there are a couple more additions to this PR I am hoping to get to over the break

jrgissing committed:
Merge branch 'lammps:develop' into bond_react_fixes_aug23

jrgissing committed:
bug fix

for when reaction site has angles, but post-reaction template has none (same for dihedrals, impropers)

jrgissing committed:
Merge branch 'lammps:develop' into bond_react_fixes_aug23

jrgissing committed:
update preferred contact info

jrgissing committed:
additional check/warning for valid templates

jrgissing committed:
tiny_epoxy example correction

not sure why this issue showed up in recent LAMMPS versions

jrgissing committed:
Merge branch 'lammps:develop' into bond_react_fixes_aug23

jrgissing committed:
remove unnecessary newton_bond considerations

related issue: fix bond/react may need modifications to work as expected when certain special_bond settings are used or when specials are defined explicitly in molecule templates and do not include all neighbors

jrgissing committed:
Merge branch 'lammps:develop' into bond_react_fixes_aug23
